pkg/cli/environment.go

```diff
@@ -93,3 +100,23 @@ func (s *EnvSettings) EnvVars() map[string]string {
 		"HELM_REPOSITORY_CONFIG": s.RepositoryConfig,
 	}
 }
+
+//Namespace gets the namespace from the configuration
+func (s *EnvSettings) Namespace() string {
+	if s.namespace != "" {
+		return s.namespace
+	}
+
+	if ns, _, err := s.KubeConfig().ToRawKubeConfigLoader().Namespace(); err == nil {
+		return ns
+	}
+	return "default"
+}
+
+//KubeConfig gets the kubeconfig from EnvSettings
+func (s *EnvSettings) KubeConfig() genericclioptions.RESTClientGetter {
+	configOnce.Do(func() {
```

Comment:
Okay, one last thing I'm noticing as I'm thinking through this design.

Given the following use case:

```
s := New()
s.KubeContext = "foo"
s2 := New()
s2.KubeContext = "bar"
...
// insert other setup code required here
```

With this in mind, I'd expect that the following would result in two unique kubeconfigs:

```
s.KubeConfig()
s2.KubeConfig()
```

Instead, what we'll end up with is the same kube config. This is because they are both using shared global flags.

Can we make `configOnce` and `config` part of the EnvSettings object as well? Thanks!